src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendApplicationIssueNotifications.java

```diff
@@ -17,28 +18,22 @@
     @Autowired
     private ApplicationIssuedNotification applicationIssuedNotification;
 
+    @Autowired
+    private ApplicationIssuedOverseasNotification applicationIssuedOverseasNotification;
+
+    @Autowired
+    private NotificationDispatcher notificationDispatcher;
+
     @Override
     public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
 
-        if (!caseDetails.getData().getApplication().isSolicitorApplication()) {
-
-            final CaseData caseData = caseDetails.getData();
-            final Long caseId = caseDetails.getId();
-
-            if (caseData.getApplicationType().isSole()) {
-                applicationIssuedNotification.sendToSoleApplicant1(caseData, caseId);
-                if (isNotBlank(caseData.getApplicant2EmailAddress())) {
-                    applicationIssuedNotification.sendToSoleRespondent(caseData, caseId);
-                }
-                if (caseDetails.getState() == AwaitingAos && caseData.getApplicant2().isBasedOverseas()) {
-                    applicationIssuedNotification.notifyApplicantOfServiceToOverseasRespondent(caseData, caseId);
-                }
-            } else {
-                applicationIssuedNotification.sendToJointApplicant1(caseData, caseId);
-                if (isNotBlank(caseData.getApplicant2EmailAddress())) {
-                    applicationIssuedNotification.sendToJointApplicant2(caseData, caseId);
-                }
-            }
+        final CaseData caseData = caseDetails.getData();
+        final Long caseId = caseDetails.getId();
+
+        notificationDispatcher.send(applicationIssuedNotification, caseData, caseId);
+
+        if (caseDetails.getState() == AwaitingAos) {
+            notificationDispatcher.send(applicationIssuedOverseasNotification, caseData, caseId);
```

Comment:
I am not sure how this notification works if the state is in AwaitingAos we send overseas application issued notification 🤔

Reply:
should it be not something like `caseData.getApplicant2().isBasedOverseas()` check ?

Reply:
I moved the overseas check inside the the notiification.  You are right, it might be better here